### PR TITLE
Reland "Roll engine to version e3687f70c7ece72000b32ee1b3c02755ba5361ac (#20427)"

### PR DESCRIPTION
**Pull request by mkustermann**
This is a re-land with downgraded `package:flutter_gallery_assets` version, which appears to have a weired sdk constraint allowing this particular dart sdk to be compatible, namely it uses `>=1.19.0 <2.1.0`

### CODE REVIEW

**Review by mraleph — APPROVED** [review]

File: examples/flutter_view/pubspec.yaml [hunk 1]
```diff
@@ -1,18 +1,22 @@
 name: flutter_view
 description: A new flutter project.
 
+environment:
+  # The pub client defaults to an <2.0.0 sdk constraint which we need to explicitly overwrite.
```
Comment (aam) [inline examples/flutter_view/pubspec.yaml]: This is really strange we have a constraint like that. Could it be that it is 'older' pub clients that default to <2.0.0 sdk constraint, but we should not be relying on them?

Comment (mkustermann) [reply examples/flutter_view/pubspec.yaml]: This was added [here](https://dart-review.googlesource.com/c/sdk/+/30961/5/CHANGELOG.md#129):

"""
There is now a default SDK constraint of `<2.0.0` for any package with no
existing upper bound. This allows us to move more safely to 2.0.0. All new
packages published on pub will now require an upper bound SDK constraint so
future major releases of Dart don't destabilize the package ecosystem.
"""

Comment (aam) [reply examples/flutter_view/pubspec.yaml]: Ah, good to know. Thanks!

**Review by tvolkert — APPROVED** [review]
LGTM

**Comment by Hixie** [comment]
This regressed memory usage on the flutter_gallery__memory_nav benchmark quite severely.

**Comment by mkustermann** [comment]
@Hixie After the roll I've checked the benchmarks and filed https://github.com/flutter/flutter/issues/20690. The suspicion so far seems this was caused by engine change https://github.com/flutter/engine/commit/149ea7b74c55344c3a97fdf9ca6c88217abf92b9 (see the issue)

**Pull request merged**